Development - Pull Request 12Jun2020

charre2017idv opened the pull request:
This pull request contains the next information:

- [ ] buMath.h 
- [x] buPlatformMath.h - This is for the basic math 
- [x] Plugger
- [x] Degrees
- [x] Radians
- [x] Vector2I - **Unit testing implemented**
- [x] Vector2f - **Unit testing implemented**
- [x] Vector3I - **Unit testing implemented**
- [x] Vector3f - **Unit testing implemented**
- [x] Vector4I - **Unit testing implemented**
- [x] Vector4f - **Unit testing implemented**
- [x] Matrix4x4 - **Unit testing implemented**
- [x] Quaternions - **Unit testing implemented** - _Need to check what other vector classes can be used_
- [ ] Shapes - Is the father of the next classes
  - [x] Rectangle - **Unit testing implemented**
  - [x] Box - **Unit testing implemented**
  - [x] Planes - **Unit testing implemented**
  - [x] Spheres - **Unit testing implemented** - _Need to check equation._ 
  - [X] Rays - **Unit testing implemented**
  - [ ] Frustrum

robertocharretonkaplun committed:
Merge branch 'master' into dev

robertocharretonkaplun committed:
dev - Add Doxygen Documentation

robertocharretonkaplun committed:
dev - New header implementations

This headers were added to the Engine:
- buString.h
- buStdHeaders.h
- buPrerequisitsUtil.h
- buPlatformTypes.h

robertocharretonkaplun committed:
dev - Math lib initialization

The vectors for 2D, 3D and 4D are create in int32 and floats

robertocharretonkaplun committed:
dev - Math principal methods

Principal methods for the buMath class and buPlan initialization.

robertocharretonkaplun committed:
dev - Unit Testing for vectors

There is a bug in the magnitud of the buVector3F

robertocharretonkaplun committed:
dev - Documentation + unit testing

robertocharretonkaplun committed:
dev - Matrix class + determinant unit testing

The matrix class still in process of development.

robertocharretonkaplun committed:
dev - Matrix addition

robertocharretonkaplun committed:
dev - Matrix subtraction, multiplication and other stuff

robertocharretonkaplun committed:
dev - Quaternion implementation

Basic method for initialization, adding and subtracting quaternions.

robertocharretonkaplun committed:
dev - Queaternions and other struff

Chenge the methods from buMath to buMathPlatform.
- Add a new class for rays and fix some issues with the unit testing

robertocharretonkaplun committed:
dev - Shape mathematics

- Add sphere.
- Add box.
- Add rays.
- Fix buPlatformMath.

robertocharretonkaplun committed:
dev - Rectangle shape math

robertocharretonkaplun committed:
dev - Radians + Degrees and plugman

robertocharretonkaplun committed:
dev - Change on operator return data

NeonGE commented on buUtilities/include/buBox.h:
```diff
@@ -0,0 +1,64 @@
+		/**
+		 * @brief 
+		 */
+		float m_length;
```
Este tipo de estructura requiere posiciones, no solo tamaño

charre2017idv replied:
Ya se agrego posicion.

NeonGE commented on buUtilities/include/buDegrees.h:
```diff
@@ -0,0 +1,47 @@
+		 * @brief Copy constructor
+		 */
+		buDegrees(const buDegrees& _d);
+		/**
```
Falta creacion con Radianes

charre2017idv replied:
Listo!

NeonGE commented on buUtilities/include/buDegrees.h:
```diff
@@ -0,0 +1,47 @@
+		 * @bug None.
+		 */
+		float
+		getRadians();
```
Estandariza los getters en ambas clases

charre2017idv replied:
Listo!

NeonGE commented on buUtilities/include/buMath.h:
```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include "buPrerequisitesUtil.h"
+
```
Faltan includes de sistema

NeonGE commented on buUtilities/include/buMatrix4x4.h:
```diff
@@ -0,0 +1,142 @@
+		 * @return None.
+		 * @bug None.
+		 */
+		void
```
Esto debería de regresar la dirección de este objeto

charre2017idv replied:
Ya se corrigio este metodo.

NeonGE commented on buUtilities/include/buMatrix4x4.h:
```diff
@@ -0,0 +1,142 @@
+		 * @return None.
+		 * @bug None.
+		 */
+		void
```
Esto debería de regresar la dirección de este objeto

charre2017idv replied:
Ya se corrigio este metodo

NeonGE commented on buUtilities/include/buMatrix4x4.h:
```diff
@@ -0,0 +1,142 @@
+		 * @bug None.
+		 */
+		float 
+		det();
```
Usa nombres de función descriptivos

charre2017idv replied:
Ya se corrigio el nombre de este metodo.

NeonGE commented on buUtilities/include/buPlugger.h:
```diff
@@ -0,0 +1,57 @@
+/**
```
Cambia a Pluggin

charre2017idv replied:
Ya se cambio el nombre de la clase.

NeonGE commented on buUtilities/include/buPlugger.h:
```diff
@@ -0,0 +1,57 @@
+
+	inline bool buPlugger::CreatPlugIn(LPCSTR _dllName)
+	{
+		HINSTANCE hInstance = LoadLibraryExA(
```
La instancia debe ser miembro de la clase

charre2017idv replied:
Ya se creo un miembro de la instancia.

NeonGE commented on buUtilities/include/buMatrix4x4.h:
```diff
@@ -0,0 +1,142 @@
+		/**
+		 * @brief Member in charge of giving an integer point with zero values.
+		 */
+		static const buMatrix4x4 Zero;
```
Constantes en mayusculas

charre2017idv replied:
Ya se cambio a mayusculas esta constante

NeonGE commented on buUtilities/include/buRectangle.h:
```diff
@@ -0,0 +1,63 @@
+		getPerimeter();
+
+	public:
+		/**
```
Requiere posicion

charre2017idv replied:
Ya se agrego una posicion a esta clase.

NeonGE reviewed (changes requested):
Tírame toda la documentación en HTML
Revisa los comentarios, aplican para todo el código
Elimina los ZIP

robertocharretonkaplun committed:
dev - Pull Request changes

All the notes that Prince specify are done, except from the system includes in buMath.h

robertocharretonkaplun committed:
dev - Add taylor equation

robertocharretonkaplun committed:
dev - Fix taylor int to float issues

robertocharretonkaplun committed:
dev - Updated documentation file

The documentation file change and has a new destination  path.

robertocharretonkaplun committed:
dev - Core project initializations

**Added:**
- "buCore" Project
- "buCore_UnitTest" Project
- "buDX_API" Project
- "buDX_API_UnitTest" Project

robertocharretonkaplun committed:
dev - DX initialization

robertocharretonkaplun committed:
dev - Lib correction path

robertocharretonkaplun committed:
dev - Changes

## v1.0.0 Jul. 10, 2020 
 
**Added:** 
- "BuCoreBuffer" Class 
- "BuDXBuffer" Class 
 
## v1.0.0 Jul. 09, 2020 
 
**Added:** 
- "buCoreInputLayout" class 
- "buDXInputLayout" class 
 
## v1.0.0 Jul. 08, 2020 
Engine initialization to render a triangle 
 
**Added:** 
- "buCoreTexture2D" Class 
- "buDXTexture2D" Class 
- "buCoreRenderTargetView" Class 
- "buDXRenderTargetView" Class 
- "buCoreViewport" Class 
- "buDXViewport" Class 
- "buCoreVertexShader" Class 
- "buDXVertexShader" Class 
- "buCorePixelShader" Class 
- "buDXPixelShader" Class 
- Rendering a basic triangle in Direct X 11

robertocharretonkaplun committed:
dev - Rendering the cube

robertocharretonkaplun committed:
dev - Changle Log 21Jul2020

robertocharretonkaplun committed:
dev - Warning errors fixed

## v1.0.0 Jun. 22, 2020 
**Added:** 
- Fix warnings form the project.

robertocharretonkaplun committed:
dev - Resaze base initialization

robertocharretonkaplun committed:
dev - Fix bin compilation error

NeonGE committed:
Revision de codigo segundo parcial
25/07/2020

NeonGE committed:
Small style changes

robertocharretonkaplun committed:
dev - Fix transpose matrix
